RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,218 @@
+| node_threshold          | `u32` | The total number of validator nodes that can register to execute instructions for this asset                                      |
+| minimum_collateral      | `u64` | The minimum amount of Tari a validator node must put up in collateral in order to execute instructions for this asset.([RFC-???]) |
+| node_selection_strategy | `u32` | The selection strategy to employ allowing nodes to register to manage this asset                                                  |
+
+
+#### Issuer
+
+Anyone can issue assets on the Tari network from their wallet or client program. The wallet will provide the PKH and
+sign the instruction. The wallet needn’t use the same private key each time.
+
+#### Name and Description
+
+These fields are purely for informational purposes and do not need to be unique, and do not act as an asset ID.
+
+#### Top level digital issuers
+
+If it is likely that a digital asset issuer will be issuing many assets on the Tari Network (hundreds, or thousands),
+the issuer should strongly consider registering a name (e.g. `TARILABS`) as a top-level digital asset issuer. This is
+done via a special transaction on the base layer, described in [RFC-???](). A registered TLDI prevents spoofing of
+assets from copy cats or other malicious actors. It also makes asset discovery simpler.
+
+TLDI owners need to provide a valid signature proving that they own the given domain when creating assets.
+
+#### Asset identification
+
+Assets are identified by `<TLDI>.#_hhhhh` on the Tari network, where `#` is the template number, and `hhhhh`
+is a hash of the asset creation instruction.
+
+This allows assets to be deterministically identified from their initial state. Two different creation instructions
+leading to the same hash refer to the same single asset, by definition. Validator Nodes maintain an index of assets and
+their committees, and so can determine whether a given asset already exists; and MUST reject any `create_asset`
+instruction for an existing asset.
+
+#### Template number
+
+Tari uses templates to define the behaviour for its smart contracts. The template number refers to the type of digital
+asset begin created. The logic and rules associated with each template will be part of every[*] validator node's binary.
+
+[*]: # 'Except possibly for proprietary plugin templates running on permissioned committees'
+
+The template number is a 64-bit unsigned integer and has the following format, with 0 representing
+the least significant bit.
+
+| Bit range | Description                            |
+|:----------|:---------------------------------------|
+| 0 - 31    | Template number (0 - 4,294,967,295)    |
+| 32 - 47   | Template version (0 - 65,535)          |
+| 48 - 58   | Reserved  (Must be 0)                  |
+| 59        | Non-public nodes flag                  |
+| 60        | Confidentiality data flag              |
```

Comment:
I dont think we can ever have true confidential assets if you run in selected, assigned or what ever mode. For confidential you need nominated nodes. 

But we can use this as a way to let the VN know the asset issuer wants this private, we can then perhaps build up a naughty list of known vn's who ignore this flag.

Reply:
I agree, @hansieodendaal. Particularly around defining what we mean by confidentiality. That discussion is not for this RFC though

Reply:
Agreed @cjs77, so we can just have the flag for when we can do something about it. Maybe it will mean partial confidentiality vs. none at all, or similar.